src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs

```diff
@@ -32,37 +58,86 @@
+                   :icon :plus
+                   :onClick
+                   (fn []
+                     (let [workspace (:workspace props)
+                           dest-name-value (.-value
+                                             (.getDOMNode (@refs "destinationName")))
+                           dest-namespace-value (.-value (.getDOMNode
+                                                           (@refs "destinationNamespace")))
+                           dest-workspace-name (get workspace "name")
+                           dest-workspace-namespace (get workspace "namespace")
+                           methodRepoQuery (create-post-data
+                                             (:selected-conf @state)
+                                             dest-name-value
+                                             dest-namespace-value
+                                             dest-workspace-name
+                                             dest-workspace-namespace)
+                           url-method :post
+                           url-path (create-path
+                                      dest-workspace-namespace
+                                      dest-workspace-name)
+                           url-data (utils/->json-string methodRepoQuery)
+                           url-canned-response {:status 201
+                                                :delay-ms (rand-int 2000)}
+                           url-on-done-func (fn [{:keys [success? xhr]}]
+                                              (if success?
+                                                (utils/rlog  (str "success in import! "
+                                                               "Here, perhaps trigger re-render "
+                                                                "of the MCs in the current "
+                                                                "workspace?"))
+                                                (let [returned-err-msg
+                                                      {:message (.-statusText xhr)}]
+                                                  (js/alert
+                                                    (str "Error in Method Configuration Import : "
+                                                      returned-err-msg)))))]
+                       (swap! state assoc :show-import-mc-modal? false)
+                       (utils/ajax-orch
+                         url-path
+                         {:url url-path
+                          :method url-method
+                          :data url-data
+                          :canned-response url-canned-response
+                          :on-done url-on-done-func})))}]
+                 [comps/Button
+                  {:title-text "cancel import"
+                   :icon :x
+                   :onClick (fn []
+                              (swap! state assoc :show-import-mc-modal? false))}]]])
+    :show-when true}])
```

Comment:
You mean ":show-when true"?  It's because of the way the modal-dialog content and calling that I did.  I had it another way before ("show-when :show-import-mc-modal?"), then David suggested to put "show-when true" along with other refactorings.